### PR TITLE
cache message lists (folders) in local storage

### PR DESCRIPTION
**Pull request by ChristophWurst**
partially implements #480 

IRC log:
> jancborchardt: christophwu: the only weird thing is that it still takes long switching folders (and accounts). Can’t we show the data from the cache?

@jancborchardt this PR should speed up switching folders ;-)

### CODE REVIEW

**Comment by wurstchristoph** [comment]
I will update the background updating mechanism once PR #763 is finished, for now I simply trigger the folder reload event to fetch any changes.

**Comment by jancborchardt** [comment]
There are some timing issues when clicking on other folders while the previous one is still loading. ;) Same story as in the past – best just cancel the previous request and replace the message list directly when a new folder is clicked.

**Comment by wurstchristoph** [comment]
There are two timing issues as far as I can tell
- [x] requests triggered because you click a folder that is not cached yet
- [ ] requests triggered because a cached folder is clicked and the folder is being refreshed

To solve the second scenario there are two possible solutions:
- rework how folder lists get updates (is currently done inside the view?!)
- don't trigger folder refreshing when loading a cached message list (easy one)

I prefer the latter as I would like to include the horde sync stuff once it's ready which should replace folder updating anyway. For now this would mean if you switch folders, you would have to manually refresh it by clicking the button.
@jancborchardt  what do you think?
The first one would need another PR because this one is already pretty big. The second one could be done here (would actually make the PR smaller) ;-)

**Comment by jancborchardt** [comment]
Hmm … needing to manually refresh the folder after switching to it sounds veeeery ugly. I would even say it’s better if we manually fire a click on the »Check for new messages« via JS after switching the folder.

**Comment by wurstchristoph** [comment]
@jancborchardt alright, I updated the PR. Please review again, I hope I found and fixed all timing issues. Folders refresh is automatically triggered when you switch folders (regardless of the folder being cached or not) ;-)

**Comment by jancborchardt** [comment]
Crazy shit! :+1: 

Another review please @irgendwie @DeepDiver1975 @aidanamavi @LukasReschke :)

**Comment by jancborchardt** [comment]
@ChristophWurst needs rebase! :)

After that, a second review please homies!

**Comment by ChristophWurst** [comment]
@jancborchardt done ;-)

**Comment by ChristophWurst** [comment]
Rebased, so anybody can test it with the unified inbox :-)

File: js/views/message.js [hunk 1]
```diff
@@ -277,30 +277,16 @@ views.Messages = Backbone.Marionette.CompositeView.extend({
...
-					'from': from,
-					'to': from + 20
-				});
-		}
 		var self = this;
-		$.ajax(url, {
-				data: {},
-				type:'GET',
-				success: function (jsondata) {
+		Mail.Communication.fetchMessageList(
```
Comment (DeepDiver1975) [inline js/views/message.js]: with respect to modularity and testability (one day - LOL) I guess we should inject the Communications object into the view ..... one day ...

**Comment by jancborchardt** [comment]
So what’s the status here? Has been hanging around for 2 weeks now without new reviews. @DeepDiver1975 @Xenopathic @aidanamavi @irgendwie please check! :)

**Pull request merged**